fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/api/GlobalConfigurationApiResource.java

```diff
@@ -156,4 +156,29 @@ public String updateConfiguration(@PathParam("configId") @Parameter(description
 
         return this.toApiJsonSerializer.serialize(result);
     }
+
+    @PUT
+    @Path("/name/{configName}")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    @Operation(summary = "Update Global Configuration by name", description = "Updates an enable/disable global configuration item by name")
+    @RequestBody(required = true, content = @Content(schema = @Schema(implementation = GlobalConfigurationApiResourceSwagger.PutGlobalConfigurationsRequest.class)))
+    @ApiResponses({
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = GlobalConfigurationApiResourceSwagger.PutGlobalConfigurationsResponse.class))) })
+    public String updateConfigurationByName(@PathParam("configName") @Parameter(description = "configName") final String configName,
+            @Parameter(hidden = true) final String apiRequestBodyAsJson) {
+
+        // TODO: Would be better to support string based identifier in Commands and resolve the entity by name in the
+        // service
+        final GlobalConfigurationPropertyData configurationData = this.readPlatformService.retrieveGlobalConfiguration(configName);
+
+        final CommandWrapper commandRequest = new CommandWrapperBuilder() //
+                .updateGlobalConfiguration(configurationData.getId()) //
```

Comment:
As an extra thing, since we're using the name in the URL, I'd actually start restricting the global configuration names to a specific pattern. It must not start with `-` or `_` but with a lowercase letter. Overall, the only characters allowed should be lowercase letters and `-`, `_`.

And since these global config names are decided when developing, this validation should be part of the build/startup process.

Reply:
I tend to agree, but i am hesitant to make this validation as part of startup. Anyone could have already added a global configuration entry with underscore and for them this change is quite breaking... I would rather just update the validation when entry is created or updated and enforce there only.

Reply:
Resolved